comp/otelcol/ddprofilingextension/impl/extension.go

```diff
@@ -87,13 +107,39 @@ func (e *ddExtension) startForOCB() error {
 		profiler.WithAPIKey(string(e.cfg.API.Key)),
 	)
 
-	// todo(mackjmr): add datadogexporter hostmetadata provider to retrieve hostname, and
-	// pass it to profiler via profiler.WithHostname(). This requires a refactor in contrib,
-	// as the logic lives within an internal package.
 	if string(e.cfg.API.Site) != "" {
 		profilerOptions = append(profilerOptions, profiler.WithSite(string(e.cfg.API.Site)))
 	}
 
+	source, err := e.sourceProvider.Source(context.Background())
+	if err != nil {
+		return err
+	}
+
+	var tags strings.Builder
+	tags.WriteString(fmt.Sprintf("agent_version:%s", version.AgentVersion))
+	if e.cfg.ProfilerOptions.Env != "" {
+		tags.WriteString(fmt.Sprintf(",default_env:%s", e.cfg.ProfilerOptions.Env))
+	}
+
+	if source.Kind == "host" {
+		profilerOptions = append(profilerOptions, profiler.WithHostname(source.Identifier))
+		tags.WriteString(fmt.Sprintf(",host:%s", source.Identifier))
+	}
+
+	if source.Kind == "task_arn" {
```

Comment:
I'm not sure if this works exactly, `task_arn` can be a tag on profiles emitted from classic ECS clusters (not Fargate ECS). We rely on `r.conf.FargateOrchestrator` in our agent code (see https://github.com/DataDog/datadog-agent/blob/067029a5c7b6ca88688f52be8ec68618a6313874/pkg/trace/api/profiles.go#L82) and i'm not sure if this is available here. I believe under the hood, this relies on env vars (https://github.com/DataDog/datadog-agent/blob/067029a5c7b6ca88688f52be8ec68618a6313874/pkg/util/fargate/detection.go#L21).

Reply:
Ok - it's probably worth renaming that field since it is common to have non-Fargate with `task_arn` (classic ECS). But I understand that it's what you have to use in this case, so that should be a separate issue/PR probably.

Reply:
Agreed, will rename once I add EKS fargate support